### DISCUSSION TITLE
TouchEffect.Command not working in UWP

### DISCUSSION THREAD

**Discussion by BelightWavre** [xamarin/XamarinCommunityToolkit#1607] [Q&A] [answered]
I can't succeed to make the xct:TouchEffect.Command working on UWP while:
 - it is working for android in the same project
 - there is no build error or warning

I made a very small project to test it:
1. Start a new Mobile APP project in VS2019
2. Add the Xamarin.CommunityToolKit
3. Add two properties in the Mainpage.xaml.cs (ok, it could be better implemented but it works) :

```
        private int count = 10;
        public  int Count
        {
            get => count;
            set
            {
                count = value;
                OnPropertyChanged(nameof(Count));
            }
        }

        public ICommand PressedCommand { get => new Command ( () => Count++ ); }
```

4. Update the MainPage.xaml like this:

```
     <?xml version="1.0" encoding="utf-8" ?>
     <ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
                  xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
                  xmlns:xct="http://xamarin.com/schemas/2020/toolkit"
                  x:Class="CommunityToolKit.MainPage"
                  x:Name="this">
 
         <StackLayout>
 
             <Frame BackgroundColor="#2196F3" Padding="24" CornerRadius="0">
                 <Label Text="Welcome to Xamarin.Forms!" HorizontalTextAlignment="Center" TextColor="White" FontSize="36"/>
             </Frame>
             <StackLayout BackgroundColor="IndianRed"
                          xct:TouchEffect.Command="{Binding Path=PressedCommand, Source={x:Reference this}}">
                 <Label Text="Start developing now" FontSize="Title" Padding="30,10,30,10"/>
                 <Label Text="Make changes to your XAML file and save" FontSize="16" Padding="30,0,30,0"/>
             </StackLayout>
 
             <Label Text="{Binding Path=Count, Source={x:Reference this}}" Padding="30" HorizontalOptions="Center"/>
             <Button Text="Click Me" Command="{Binding Path=PressedCommand, Source={x:Reference this}}"/>
         </StackLayout>
 
     </ContentPage>
```

The binding for the command of the button is working on both UWP and Android projects. But the binding of the touchEffect applied on the StackLayout is not working on the UWP project (When I click on the IndianRed Background or wherever in the StackLayout).

I mention that I'm aware of this post: https://github.com/xamarin/XamarinCommunityToolkit/issues/1456
So my options are curently the folowing:
![image](https://user-images.githubusercontent.com/55940243/131468985-4398bbd4-788d-4280-8178-9061260aef22.png)
Version of Xamarin 5.0.0.2012

What did I miss ?!? Any idea ?

**Comment by AndreiMisiukevich** [comment]
@pictos could you please check this?

**Comment by AndreiMisiukevich** [comment]
@BelightWavre does it work for you if you download this sample https://github.com/xamarin/XamarinCommunityToolkit/issues/1456#issuecomment-873653567

And do this https://github.com/xamarin/XamarinCommunityToolkit/issues/1456#issuecomment-873674900

?

**Reply by BelightWavre** [reply]
@AndreiMisiukevich, I can't download any file there. They have been deleted, I guess.

Someone talk me about the need for Xamarin.CommunityToolKit.Markup. Correct ? Needed ? How to reference / use it ? I never read something about it. I'm surprised that my code works in Android without it but should not works in UWP ...
I made a public repo with my project here :  [https://github.com/BelightWavre/CommunityToolKit](url)

**Comment by BelightWavre** [comment, marked as answer]
I found something. It works in debug mode, but not in release mode (same project - build from scratch). Does it reveal something to you ? FYI, it is the same with the release 1.0.0 and 1.2.0.

**Reply by AndreiMisiukevich** [reply]
@BelightWavre try this 
```csharp 
var assembliesToInclude = new List<Assembly>() { typeof(Xamarin.CommunityToolkit.Effects.TouchEffect).GetTypeInfo().Assembly }; 
Xamarin.Forms.Forms.Init(e, assembliesToInclude);
```

**Reply by BelightWavre** [reply]
Hello, this solves the problem. I suggest to add it to the documentation ...
It works for me with release 1.2.0 and Xamarin 5.0.0.2083.

By the way, do I need to add some other assembly for other functionalities of the Toolkit ... Before I face other problem like this ;-)

**Reply by AndreiMisiukevich** [reply]
Nope) all classes are in the same assembly)

**Reply by beaver316** [reply]
How can I do this in the iOS version? I have the same issue for iOS but the Xamarin.Forms.Forms.Init(e, assembliesToInclude); has an error.

**Reply by Ali-YousefiTelori** [reply]
> @BelightWavre try this
> 
> ```cs
> var assembliesToInclude = new List<Assembly>() { typeof(Xamarin.CommunityToolkit.Effects.TouchEffect).GetTypeInfo().Assembly }; 
> Xamarin.Forms.Forms.Init(e, assembliesToInclude);
> ```

@AndreiMisiukevich In my case, it's not working in UWP but works in android.
![Screenshot 2022-07-18 172701](https://user-images.githubusercontent.com/5262218/179516037-a9feaa0f-32b3-4018-825d-0498bbd42b5c.png)
